### ISSUE TITLE
Not possible connect Sixfab IoT shield to raspberry pi through serial port

### ISSUE THREAD

**Issue by oscramgon** [sixfab/Sixfab_RPi_CellularIoT_App_Shield#6]
Hello,

I have since October 2018 a Sixfab IoT shield. I have been able to connect shield to raspberry pi with PPP, and get IP adress in Rasapberry Pi, being able to control remotely a robot connected to this raspberry pi through Sixfab shield. I am able to read with python sensors in shield, but when I try to send AT commands from raspberry pi using python to IoT shield to get RSRP, I am getting an error like one below:

>>> from cellulariot import cellulariot
>>> import time
>>> node = cellulariot.CellularIoT()
Sixfab Cellular IoT Hat Class initialized!
>>> node.sendATComm("ATE1","OK\r\n")
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/serial/serialposix.py", line 265, in open
    self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
OSError: [Errno 16] Device or resource busy: '/dev/ttyS0'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/sixfab_cellulariot-1.0-py3.5.egg/cellulariot/cellulariot.py", line 152, in sendATComm
  File "/usr/local/lib/python3.5/dist-packages/sixfab_cellulariot-1.0-py3.5.egg/cellulariot/cellulariot.py", line 142, in sendATCommOnce
  File "/usr/lib/python3/dist-packages/serial/serialposix.py", line 268, in open
    raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg))
serial.serialutil.SerialException: [Errno 16] could not open port /dev/ttyS0: [Errno 16] Device or resource busy: '/dev/ttyS0'
>>> quit(9
... quit()
  File "<stdin>", line 2
    quit()
       ^
SyntaxError: invalid syntax

Now I am trying to connect Sixfab with Raspberry Pi through USB cable, using usb connection in IoT shield, and one out of four usb ports in raspberry pi, but I do not know how to get usb port in /dev directory of raspberry pi. I am able to connect IoT shield to windows computer through usb cable, and opening a putty serial console I can reach serial port of IoT shield and send AT commands and receive results from IoT shield. At this point I have connected Iot shield with raspberry pi only through usb cable, without insert Iot shield on 40 pins connection of raspberry pi.

I guess that I need drivers in raspberry pi in order to permit to Pi read serial port through usb, but following PDF Quectel_WCDMA&LTE_Linux_USB_Driver_User_Guide_V1.8.pdf is confusing for me since I do not know how to download kernel files.

Please, could you give me some clue about how to follow with this config and be able to connect Pi and Shield through usb serial cable?

Thank you for your help and best regards
Oscar

**Comment by saeedjohar** [comment]
The issue over here is you are already connected to PPP via ttyS0. In this case you won't be able to send AT commands via S0. You can use a short USB cable for your shield and Raspberry Pi. Use ttyUSB2 to send AT commands.

In this case, you can create your instance as
node = cellulariot.CellularIoT(serial_port="/dev/ttyUSB2")

Thank you.

**Comment by youngie** [comment]
This product is crap and their support is even worse.I spent the good part of two months on my own time, at my own expense working with Saeed at Sixfab trying to get this POS to work and solve their issues with their CRAPTEL shitty hardware. I incurred out of pocket expenses that were never reimbursed. I tried multiple boards, multiple locations, drove their shit all over town as they thought it was a signal issue, issued a million AT commands and sent them all the logs. We tried multiple antennas, different cell phone carriers. As a software developer with well over 25 years of experience I would say stay well clear of this rubbish!
